lighthouse-core/test/lib/drivers/driver-test.js

```diff
@@ -80,4 +83,123 @@ describe('Browser Driver', () => {
     assert.notEqual(opts.url, req1.url, 'opts.url changed after the redirects');
     assert.equal(opts.url, req3.url, 'opts.url matches the last redirect');
   });
+
+  it('will fail when multiple tabs are found with the same active serviceworker', () => {
+    const pageUrl = 'https://example.com/';
+    const swUrl = `${pageUrl}sw.js`;
+    const registrations = [
+      createSWRegistration(1, pageUrl),
+    ];
+    const versions = [
+      createActiveWorker(1, swUrl, ['unique'])
+    ];
+
+    driverStub.getCurrentTabId = () => Promise.resolve('unique2');
+    driverStub.once = createOnceStub({
+      'ServiceWorker.workerRegistrationUpdated': {
+        registrations
+      },
+      'ServiceWorker.workerVersionUpdated': {
+        versions
+      }
+    });
+
+    return driverStub.checkForMultipleTabsAttached(pageUrl)
+      .then(_ => assert.ok(false), _ => assert.ok(true));
+  });
+
+  it('will succeed when service worker is already registered on current tab', () => {
+    const pageUrl = 'https://example.com/';
+    const swUrl = `${pageUrl}sw.js`;
+    const registrations = [
+      createSWRegistration(1, pageUrl),
+    ];
+    const versions = [
+      createActiveWorker(1, swUrl, ['unique'])
+    ];
+
+    driverStub.getCurrentTabId = () => Promise.resolve('unique');
+    driverStub.once = createOnceStub({
+      'ServiceWorker.workerRegistrationUpdated': {
+        registrations
+      },
+      'ServiceWorker.workerVersionUpdated': {
+        versions
+      }
+    });
+
+    return driverStub.checkForMultipleTabsAttached(pageUrl)
+      .then(_ => assert.ok(true), _ => assert.ok(false));
+  });
+
+  it('will succeed when old sw are deleted', () => {
```

Comment:
is this test still what you want given what @paulirish said about deleted service workers?